externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/TrinoStatement.scala

```diff
@@ -80,63 +74,34 @@ class TrinoStatement(
     }
   }
 
-  /**
-   * Execute sql and return ResultSet.
-   */
-  def execute(): Iterable[List[Any]] = {
-    val rowQueue = new ArrayBlockingQueue[List[Any]](MAX_QUEUED_ROWS)
-
-    val dataProcessing = Future[Unit] {
-      while (trino.isRunning) {
-        val data = trino.currentData().getData()
-        if (data != null) {
-          data.asScala.map(_.asScala.toList)
-            .foreach(e => putOrThrow(rowQueue, e))
+  def execute(): Iterator[List[Any]] = {
+    Iterator.continually {
+      @tailrec
+      def getData(): (Boolean, List[List[Any]]) = {
+        if (trino.isRunning) {
+          val data = trino.currentData().getData()
+          trino.advance()
+          if (data != null) {
+            (true, data.asScala.toList.map(_.asScala.toList))
+          } else {
+            getData()
+          }
+        } else {
+          Verify.verify(trino.isFinished)
+          val finalStatus = trino.finalStatusInfo()
+          if (finalStatus.getError() != null) {
+            throw KyuubiSQLException(
+              s"Query ${finalStatus.getId} failed: ${finalStatus.getError.getMessage}")
```

Comment:
If the user invokes the `getQueryId` interface, is this query id same as the returned id of `getQueryId`?

Reply:
yes，the `queryId` is generated by trinoServer at the time of the first query and is unique throughout the query process